tests/test_parser.py

```diff
@@ -127,26 +127,28 @@ def test_parameter(self):
     def test_var(self):
         self.assertEqual(parse_one("SELECT @JOIN, @'foo'").sql(), "SELECT @JOIN, @'foo'")
 
-    def test_annotations(self):
+    def test_comments(self):
         expression = parse_one(
             """
-            SELECT
-                a #annotation1,
-                b as B #annotation2:testing ,
-                "test#annotation",c#annotation3, d #annotation4,
-                e #,
-                f # space
+            --comment1
+            SELECT /* this won't be used */
```

Comment:
Because in the tokens module if we have 2 comments attached in _the same_ token, the trailing one is discarded. The reason is that it didn't really made sense to have two different comments attached to the same token. Also, we'd have to carry both leading and trailing comments for every token otherwise.

Reply:
but this comment should be attached to the select because it's right after it

Reply:
what if you just did

select -- coment

Reply:
If select didn't have a leading comment, that'd bind to it and it would work as expected.